### PR TITLE
010 refs

### PR DESCRIPTION
**Pull request by vEpiphyte**
replace 00x refs command with a auto-traversing storm command

### CODE REVIEW

File: synapse/lib/storm.py [hunk 1]
```diff
@@ -388,3 +394,167 @@ def runStormCmd(self, snap, genr):
...
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('-d', '--degrees', type=int, default=1, action='store',
+                          help='Number of degrees to walk from the source node.')
+        pars.add_argument('--traverse-edge', default=False, action='store_true',
+                          help='Traverse Edge type nodes, if encountered, to '
+                               'the opposite side of them, if the opposite '
+                               'side has not yet been encountered.')
+        pars.add_argument('--join', default=False, action='store_true',
+                          help='Include source nodes in the output of the refs command.')
+        pars.add_argument('--omit-traversal-form', action='append', default=[], type=str,
```
Comment (vEpiphyte) [inline synapse/lib/storm.py]: Some of these arguments are pretty long - i'm open to changing them and providing short (single character) alternatives.

File: synapse/lib/storm.py [hunk 2]
```diff
@@ -388,3 +394,167 @@ def runStormCmd(self, snap, genr):
             node = snap.getNodeByBuid(buid)
             if node:
                 yield node, node.initPath()
+
+class RefsCmd(Cmd):
+    '''
+    Get nodes adjacent to inbound nodes.
```
Comment (vEpiphyte) [inline synapse/lib/storm.py]: need better doc here

File: synapse/lib/storm.py (hunk 1 above)
Comment (whippit) [reply synapse/lib/storm.py]: +1 to having a short form and long form.

File: synapse/lib/storm.py [hunk 3]
```diff
@@ -388,3 +394,167 @@ def runStormCmd(self, snap, genr):
...
+    '''
+    name = 'refs'
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('-d', '--degrees', type=int, default=1, action='store',
```
Comment (whippit) [inline synapse/lib/storm.py]: Consider adding choices here to constrain silliness and you can remove the check in `runStormCmd`.  `choices=xrange(1, 10)` or some such

Comment (vEpiphyte) [reply synapse/lib/storm.py]: per discussion, we're allowing a user to have as much silliness as possible.

Comment (whippit) [reply synapse/lib/storm.py]: :rofl:

**Pull request merged**